evol/evolution.py

```diff
@@ -140,29 +141,30 @@ def breed(self, parent_picker, combiner, population_size=None, name=None, **kwar
                                         population_size=population_size, **kwargs))
 
     def mutate(self, func, probability=1.0, name=None, **kwargs) -> 'Evolution':
-        """Add a mutate step to the Evolution.
+        """Add a mutate_with step to the Evolution.
 
         This mutates the chromosome of each individual.
 
-        :param func: Function that accepts a chromosome and returns
+        :param mutate_func: Function that accepts a chromosome and returns
             a mutated chromosome.
-        :type func: Callable[chromosome, **kwargs] -> chromosome
+        :type mutate_func: Callable[chromosome, **kwargs] -> chromosome
         :param probability: Probability that the individual mutates.
             The function is only applied in the given fraction of cases.
             Defaults to 1.0.
         :type probability: float
-        :param name: Name of the mutate step.
+        :param name: Name of the mutate_with step.
         :type name: str
         :param kwargs: Kwargs to pass to the parent_picker and combiner.
             Arguments are only passed to the functions if they accept them.
         :return: self
         """
-        return self._add_step(MutateStep(name=name, probability=probability, func=func, **kwargs))
+        return self._add_step(MutateStep(name=name, probability=probability, mutate_func=func, **kwargs))
```

Comment:
The naming of the _function_ arguments is erratic in general. I think we should spend some time getting this right.